### ISSUE TITLE
omv fails with 'Preexistent datafile...' if parent folders contain spaces

### ISSUE THREAD

**Issue by JustasB** [OpenSourceBrain/osb-model-validation#56]
I cannot get omv to run any tests, keep getting the above message and the (observed data) array is empty (looks like it's not reading the .dat file at all). MacOS. On windows, spent significant time on resolving dependency issues, to no avail. 

Happens even if: 
-the last run was >30 mins ago
-the .dat files are deleted prior to running the test (ie. don't exist)
-omv test XYZ.omt or omv all is run

Looking at: https://github.com/OpenSourceBrain/osb-model-validation/issues/51, this appears to be a different issue as the timestamps are more than 1s apart.

Looking at the code: https://github.com/OpenSourceBrain/osb-model-validation/blob/75b91dbea9f335ee0e5feb9dc611749655395b75/omv/analyzers/spikes.py consider one of the following changes:
-Change the error to a warning and continue reading the file anyway
-Delete the existing file
-Implement a different change detection scheme (MD5 hash?)

Details:
omv test .test.fig3.jnml.omt -V
```
[omv]     ERROR! Preexistent datafile Fig3_Vs.dat has not been updated!
[omv]     ---------------------------------------------------------------------------------------------------------------------------------------------------------------------
[omv] Comparison of 
                    (observed data): []
                    and
                    (expected data): [92.77, 96.3, 102.3]
                    failed against tolerance 0
[omv]       Observable                        Test Passed     
[omv]       --------------------------------------------------
[omv]       spike times                            ✘ 
```

**Comment by borismarin** [comment]
Could you point me to the project where those problematic tests are being
made?

**Comment by JustasB** [comment]
@borismarin Keep in mind this happens only locally, travis seems to do the job fine. 
I'm trying to get this pull request to work: https://github.com/OpenSourceBrain/PinskyRinzelModel/pull/12

However, cannot get those tests to run locally. For now, resorting to checking in the changes, waiting for travis, then seeing which tests failed/passed.

**Comment by borismarin** [comment]
And is that the case only for this project, or any local run of `omv`?

**Comment by JustasB** [comment]
Only tried this project. Though there are several different neuroml projects, with different source files in different folders, within this project. Same error for all three sub-projects.

**Comment by borismarin** [comment]
![image](https://cloud.githubusercontent.com/assets/3452783/8569046/65eee850-256b-11e5-8013-42dc5568183d.png)

**Comment by borismarin** [comment]
Steps to reproduce 
```
pip install git+https://github.com/OpenSourceBrain/osb-model-validation.git
git clone git@github.com:JustasB/PinskyRinzelModel.git
cd PinskyRinzelModel
omv all
``` 
All tests run (but `.test.fig2.jnml.omt` fails due to too stringent a tolerance)

We'll have to check your local installation, as I can't reproduce it in 4 different systems (OSX Yosemite, OSX Mavericks, Ubuntu 12 and travis workers).

**Comment by pgleeson** [comment]
@JustasB try also another project with OMV tests, e.g. https://github.com/OpenSourceBrain/ACnet2. Try:

    omv test  .test.validate.omt

in the neuroConstruct/generatedNeuroML2 folder.

**Comment by JustasB** [comment]
@pgleeson ACnet2 omv runs, though it's not using any mep files. Here's the output:

```
[omv] 
[omv] Running tests defined in .test.validate.omt
[omv] ===========================================
[omv]   No mep file specified. Will only run simulation using: jNeuroML_validatev1
[omv]   Running with jNeuroML_validatev1
[omv]   Running checks for experiment: Dry run
[omv] 
[omv]       Observable                        Test Passed     
[omv]       --------------------------------------------------
[omv]       dry                                    ✔          
[omv] 
[omv]                    ===============================                    
[omv]                    Test passed: .test.validate.omt 
```

**Comment by borismarin** [comment]
@JustasB I'd like to get to the bottom of it - is it possible to ssh to your machine? if not, we can have a chat if that suits you.

**Comment by JustasB** [comment]
@borismarin @pgleeson The error message goes away, and tests run as normal, when there are no spaces in any of the parent folders.

Steps to repro:

OK:
```
pip install git+https://github.com/OpenSourceBrain/osb-model-validation.git
git clone git://github.com/JustasB/PinskyRinzelModel.git
cd PinskyRinzelModel/NeuroML2/LEMS
omv test .test.fig3.jnml.omt -V
```

Results in error:
```
mkdir test\ space
cd test\ space
pip install git+https://github.com/OpenSourceBrain/osb-model-validation.git
git clone git://github.com/JustasB/PinskyRinzelModel.git
cd PinskyRinzelModel/NeuroML2/LEMS
omv test .test.fig3.jnml.omt -V
```

**Comment by borismarin** [comment]
@JustasB Great, you found it! Shall I fix it or are you working on a PR?

**Comment by JustasB** [comment]
@borismarin No PR, please fix if you have the time. Thank you for the help!

**Comment by JustasB** [comment]
@borismarin Are you working on this? I got the debugger to load, and could fix this.

**Comment by JustasB** [comment]
This is not a problem on windows. So far, mac os, has the space issue.

**Comment by borismarin** [comment]
@JustasB sorry, I was away. The cause is a bug in `jnml`, see https://github.com/NeuroML/jNeuroML/pull/25 for the fix.

**Comment by JustasB** [comment]
@borismarin Ok, I'll leave it alone then. Thanks for the fix!

**Comment by borismarin** [comment]
@JustasB thank **you**, for first trying - and also fixing - _omv_ in window$!

**Comment by borismarin** [comment]
closed as per NeuroML/jNeuroML#25
